### ISSUE TITLE
Failed to import block on sepolia

### ISSUE THREAD

**Issue by mcdee** [paradigmxyz/reth#7138]
### Describe the bug

reth synced the chain on sepolia, but a few hours later it failed to import a block and stalled. These messages loop, and there is no progress on the sepolia chain.

### Steps to reproduce

Happened initially on beta 1, upgraded to beta 2 and problem still occurs.

### Node logs

```text
{"timestamp":"2024-03-14T08:47:52.238519Z","level":"WARN","fields":{"message":"Error while processing payload","error":"Failed to insert block (hash=0xcd492ceb4b90baee2e5fb6e7a1202ed8c6d888faa2b2eb5ef8c0fd9feb26ed9e, number=5470136, parent_hash=0x21fe5f6cef4e7f028341dc1859de3da254385389b4f3ca162e62cff481148400): mismatched block state root: got 0xf2ce548be402a285951097970f0495b2993c755246e6a76b22ff08ec0dbf3967, expected 0x4fdc390aebea912aae91b73eda62486c4ac28db8d64a5e1a69801c1c2765592c"}}
{"timestamp":"2024-03-14T08:47:52.238545Z","level":"WARN","fields":{"message":"Invalid block error on new payload","invalid_hash":"0xcd492ceb4b90baee2e5fb6e7a1202ed8c6d888faa2b2eb5ef8c0fd9feb26ed9e","invalid_number":"5470136","error":"mismatched block state root: got 0xf2ce548be402a285951097970f0495b2993c755246e6a76b22ff08ec0dbf3967, expected 0x4fdc390aebea912aae91b73eda62486c4ac28db8d64a5e1a69801c1c2765592c"}}
{"timestamp":"2024-03-14T08:47:54.045869Z","level":"INFO","fields":{"message":"Status","connected_peers":21,"freelist":"279852","latest_block":"5470135"}}
```

### Platform(s)

Linux (x86)

### What version/commit are you on?
```

reth Version: 0.2.0-beta.2
Commit SHA: 28f3a2e
Build Timestamp: 2024-03-13T17:30:56.136401700Z
Build Features: jemalloc
Build Profile: maxperf

```

### What database version are you on?
```

Current database version: 2
Local database version: 2

```

### What type of node are you running?

Full via --full flag

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by mcdee** [comment]
A resync from scratch with beta 3 fails with a similar error at 5534551.

**Comment by mcdee** [comment]
A resync from scratch with beta 4 fails with a similar error at 5576459.

**Comment by gakonst** [comment]
Could you share some more logs here? Want to understand more about whether this occurs during pipeline sync, or after transitioning to live?

**Comment by mcdee** [comment]
[reth.log](https://github.com/paradigmxyz/reth/files/14787060/reth.log)

100 lines before and 50 lines after the first instance of "Invalid" in the log, let me know if you need more.

**Comment by mattsse** [comment]
sorry about this.

this looks like a state root issue for a payload received from the engine.

I'm syncing past these blocks now, so I assume something went wrong during live sync.
given the low number of forked blocks, https://sepolia.etherscan.io/blocks_forked I think we can rule out reorg issues.

checking if I can reproduce this with a --full node, might be pruner related

**Comment by gakonst** [comment]
Fixed in #7753.
